project-api-tests/src/test/java/org/ballerina/projectapi/BalToolTest.java

```diff
@@ -0,0 +1,870 @@
+        // check for tool availability in bal-tools.toml
+        BalToolsToml balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.UPDATE).balToolsTomlPath);
+        Optional<BalToolsManifest.Tool> toolOpt = BalToolsManifestBuilder.from(balToolsToml).build()
+                .getActiveTool(toolId);
+        String toolIdAndVersion = toolId + ":" + latestVersion;
+        if (toolOpt.isEmpty()) {
+            Assert.fail("Tool '" + toolIdAndVersion + "' should be the current version in bal-tools.toml");
+        }
+
+        Process cmdExec = executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("update", toolId)),
+                toolEnvironments.get(ToolSubCommand.UPDATE).envVariables);
+        String cmdErrors = getString(cmdExec.getErrorStream());
+
+        if (!cmdErrors.isEmpty()) {
+            Assert.fail(OUTPUT_CONTAIN_ERRORS + cmdErrors);
+        }
+
+        // Validate the command output
+        String cmdOutput = getString(cmdExec.getInputStream());
+        String expectedOutput = readExpectedCmdOutsAsString("tool-update-with-no-new-version.txt");
+        if (!cmdOutput.contains(expectedOutput)) {
+            Assert.fail(OUTPUT_NOT_CONTAINS_EXP_MSG + expectedOutput + "\nactual output:" + cmdOutput);
+        }
+
+        // check for tool availability in bal-tools.toml
+        balToolsToml = BalToolsToml.from(toolEnvironments.get(ToolSubCommand.UPDATE).balToolsTomlPath);
+        toolOpt = BalToolsManifestBuilder.from(balToolsToml).build().getActiveTool(toolId);
+        toolIdAndVersion = toolId + ":" + latestVersion;
+        if (toolOpt.isEmpty()) {
+            Assert.fail("Tool '" + toolIdAndVersion + "' is not available in bal-tools.toml");
+        }
+    }
+
+    @BeforeGroups(value = "use", dependsOnGroups = {"update"})
+    public void setupUseTests() throws IOException, InterruptedException {
+        // Pull the latest tool version
+        executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("pull", toolId)), toolEnvironments.get(ToolSubCommand.USE).envVariables);
+
+        // Pull tool version with specific tool version
+        String specificToolVersion = toolId + ":" + specificVersion;
+        executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("pull", specificToolVersion)),
+                toolEnvironments.get(ToolSubCommand.USE).envVariables);
+
+        // Pull tool version with incompatible distribution
+        String incompToolVersion = toolId + ":" + incompatibleDistVersion;
+        executeToolCommand(DISTRIBUTION_FILE_NAME, tempWorkspaceDirectory,
+                new ArrayList<>(Arrays.asList("pull", incompToolVersion)),
+                toolEnvironments.get(ToolSubCommand.USE).envVariables);
+
+        // add the dist incompatible version to the toml file
```

Comment:
```suggestion
        // Add the dist incompatible version to the toml file
```

Reply:
Check for other places as well.

Reply:
Fixed with 0b92fd97f16524ed01af439c88ff2b462485cd34